### DISCUSSION TITLE
Validation.email: is john72@forum.dk:frame13531 valid email ?

### DISCUSSION THREAD

**Discussion by babacarcissedia** [laravel/framework#42739] [Polls]
Just got this on one of my websites.
![image](https://user-images.githubusercontent.com/17571380/172903676-c9fabad2-a2ba-4edf-9ff9-6b5428e729da.png)

Email not able to be sent
![image](https://user-images.githubusercontent.com/17571380/172904387-fdcab82d-0050-4610-9ffa-df742006f87a.png)

Laravel doesn't think it is
![image](https://user-images.githubusercontent.com/17571380/172904234-447095f2-7c8e-429e-85f3-8a35ce07412f.png)

Should this be a valid email `john72@forum.dk:frame13531` ? If so, should we raise this as a bug ?

PS: john72@forum.dk:frame13531 is not the real email used just for privacy

**Comment by BrekiTomasson** [comment]
The old adage "There are only two hard things in Computer Science: cache invalidation and naming things." should really be revisited, because it's obvious that there are **three** hard things in Computer Science: cache invalidation, naming things and e-mail format validation.

There have been so many attempts over the years to write a concise set of rules and regulations for what is and isn't a valid e-mail, but the only real authorities are things like [RFC 822 (section 6.1)](https://www.rfc-editor.org/rfc/rfc822#section-6.1) and so on.

My own implementation of a "valid" email address resolver looks like this:

- A "local" part (before the ´@´) that contains any ASCII character, including control characters.
- Certain ASCII characters (spaces and the following: `"(),:;<>@[\]`) in the "local" part can only be allowed inside quoted strings.
- The "local" part may not begin or end with a period (`.`)
- The "local" part may not have two periods in a row at any point, unless inside a quoted string.
- The "domain" part, in full, has to resolve to an `MX` record in a DNS lookup or, if an `MX` record is missing, an `A` record for said domain.

And in this case, while your address passes the first four rules, it fails the fifth and final:

```
❯ nslookup
> set type=mx
> forum.dk
Server:		10.0.1.2
Address:	10.0.1.2#53

Non-authoritative answer:
forum.dk	mail exchanger = 10 forummail.forum.dk.

> forum.dk:frame13531
Server:		10.0.1.2
Address:	10.0.1.2#53

** server can't find forum.dk:frame13531: NXDOMAIN
```

**Reply by babacarcissedia** [reply]
Thank you for taking the time. I may consider write my own email validation rule with these ✅

**Reply by babacarcissedia** [reply]
## Update
I found this on [laravel documentation](https://laravel.com/docs/9.x/validation#rule-email) leading to [this package](https://github.com/egulias/EmailValidator) that handles both RFC and dns validation.
This should handle most except one: that the email inbox is valid. We can check valid by looking if a message would go to that address (ie not-a-valid-gmail-address@gmail.com)
